### PR TITLE
[fix](Nereids)join output order need same with child plan node output when translate

### PR DESCRIPTION
**Pull request by morrySnow**
## Problem summary

In BE, There is an implicit convention that HashJoinNode's left child's output Slot must before right child's output slot in intermediateTuple.
However, after we do commute rule on join plan in Nereids, this convention will be broken and cause core dump in BE.

There are two way to fix this problem:
1. add a project on join after we do commute
2. reorder output of join node when we do translate

Since we cannot translate project yet because BE projection support is on going(#11842). So we use second way to fix it now. After the project translation could work correctly, we should use the first way to fix it.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by jackwener — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -363,32 +367,48 @@ public PlanFragment visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> hashJoin,
...
+
+        List<Expr> execEqConjuncts = hashJoin.getHashJoinConjuncts().stream()
                 .map(EqualTo.class::cast)
                 .map(e -> swapEqualToForChildrenOrder(e, hashJoin.left().getOutput()))
                 .map(e -> ExpressionTranslator.translate(e, context))
                 .collect(Collectors.toList());
+
+        // in Nereids, all node only has one TupleDescriptor, so we can use the first one.
+        TupleDescriptor leftTuple = context.getTupleDesc(leftPlanRoot.getTupleIds().get(0));
+        TupleDescriptor rightTuple = context.getTupleDesc(rightPlanRoot.getTupleIds().get(0));
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:387]: If Nereids only has one TupleDescriptor, we should hide the tupleId in the context, or else people who add new translate method maybe misuse？

So I suggest
```java
TupleDescriptor joinTuple = context.getTupleDesc();
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 2]
```diff
@@ -363,32 +367,48 @@ public PlanFragment visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> hashJoin,
...
+        // but BE need left child's output must be before right child's output.
+        // So we need to swap the output order of left and right child if necessary.
+        // TODO: revert this after Nereids could ensure the output order is correct.
         TupleDescriptor outputDescriptor = context.generateTupleDesc();
-        List<Expr> srcToOutput = hashJoin.getOutput().stream()
+        Map<ExprId, SlotReference> slotReferenceMap = Maps.newHashMap();
+        hashJoin.getOutput().stream()
                 .map(SlotReference.class::cast)
+                .forEach(s -> slotReferenceMap.put(s.getExprId(), s));
+        List<Expr> srcToOutput = Stream.concat(leftTuple.getSlots().stream(), rightTuple.getSlots().stream())
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:398]: if there is only one TupleDescriptor, you no need to concat slots

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**